crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -758,83 +712,92 @@ where
         match worker_state.heap.insert(heap_key) {
             // Entry is already occupied, we need to find and replace old piece with new one
             Some(KeyWrapper(old_piece_index)) => {
-                for (cache_index, cache) in caches.iter_mut().enumerate() {
-                    let old_record_key = RecordKey::from(old_piece_index.to_multihash());
-                    let Some(offset) = cache.stored_pieces.remove(&old_record_key) else {
-                        // Not this disk farm
-                        continue;
-                    };
-
-                    if let Err(error) = cache.backend.write_piece(offset, piece_index, &piece).await
-                    {
-                        error!(
-                            %error,
-                            %cache_index,
-                            %piece_index,
-                            %offset,
-                            "Failed to write piece into cache"
-                        );
-                    } else {
-                        trace!(
-                            %cache_index,
-                            %old_piece_index,
-                            %piece_index,
-                            %offset,
-                            "Successfully replaced old cached piece"
-                        );
-                        cache.stored_pieces.insert(record_key, offset);
-                    }
+                // for (cache_index, cache) in caches.iter_mut().enumerate() {
+                let old_record_key = RecordKey::from(old_piece_index.to_multihash());
+                let Some(offset) = caches.stored_pieces.remove(&old_record_key) else {
+                    // Not this disk farm
+                    warn!(
+                        %old_piece_index,
+                        %piece_index,
+                        "Should have replaced cached piece, but it didn't happen, this is an \
+                        implementation bug"
+                    );
                     return;
+                };
+                let cache_index = usize::from(offset.cache_index);
+                let piece_offset = offset.piece_offset;
+                let Some(backend) = caches.backends.get(cache_index).cloned() else {
```

Comment:
Does it need to be cloned though?

Reply:
beckend is wrapped in Arc. In this clone in order to avoid lending immutable borrowed

Reply:
Why do you need to avoid it? BTW you can leave multiple comments as "Review" if you switch to "Files changed" tab of the PR instead, it decreases number of notifications for repository maintainers greatly.

Reply:
Oh yeah, there's no need to avoid it, its a mistake. 

> BTW you can leave multiple comments as "Review" if you switch to "Files changed" tab of the PR instead, it decreases number of notifications for repository maintainers greatly.

Okay, I'll take care of that. (Also, I'm very sorry, but I submitted the commit with wrong author, can I just override that?)

Reply:
Sure, please force-push with name change and no other changes so I don't need to re-review it again